dream_demo.ipynb

```diff
@@ -0,0 +1,268 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(\"Generated bragg peak d spacings\") # The data can be accessed directly to check\n",
+    "print(instrument.sample_bragg_peaks_d)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Generating random time of flight data\n",
+    "The fake instrument and sample are now used to generate a requested number of events. \n",
+    "\n",
+    "The wavelength window can be adjusted. \n",
+    "\n",
+    "A simple text based progress bar will be shown unless the keyword argument `verbose` is set to False. \n",
+    "\n",
+    "The resulting data is checked and compared to the ESS timeframe, it would be optimal to have just under 100% coverage."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(\"Generating random sparse data\")\n",
+    "instrument.generate_data_pseudo(n_events, wavelength_width=3.5, wavelength_center=3.0)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(instrument.data) # Print an overview of the resulting dataset"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Convert time of flight data to dspacing\n",
+    "The raw time of flight data now needs to be converted back to dspacing. This can be done using the calibration data calculated within the Dream object, or without assuming a perfect instrument. \n",
+    "\n",
+    "The two methods will give the same results in this case."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dspacing = powder_reduction.powder_reduction(instrument.data, calibration=instrument.cal)\n",
+    "print(dspacing) # Show the converted dataset\n",
+    "print(dspacing.dims) # Show the dimensionality of the dspacing dataset"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Plotting the dspacing dataset\n",
+    "Before plotting, the sparse data is converted to histograms for each detector. A few of these are plotted, and it is clear they collect different parts of the dspace range due to their different 2theta. Then the histograms for all detectors are summed and the final dpsacing histogram is plotted."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "N=501\n",
+    "bins = sc.Variable([Dim.DSpacing],\n",
+    "                   values=np.linspace(2.0, 10.0, N).astype(np.float64),\n",
+    "                   unit=sc.units.angstrom)\n",
+    "\n",
+    "hist_data = sc.histogram(dspacing, bins)\n",
+    "print(hist_data)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%matplotlib inline\n",
```

Comment:
This should not be needed. `scipp.plot` should setup things correctly. It works for me without, does it not for you?

Reply:
The %matplotlib inline was requested by a reviewer who couldn't run without it, I did not have issues. I suggest we leave it for wider compatibility.

Reply:
I suggest we track down the issue. How did the reviewer run it?